### PR TITLE
Fix/recursive methods

### PR DESCRIPTION
**Pull request by LuisTejedaS**
## Description

Fixes infinite loop for different scenarios in conversion:

XSD with:
Groups that have the next dependencies: A-B-C-D-A
Elements that have the next dependencies  A-B-C-D-A

Fixes infinite loop for scenario in merging:
Added scenario for circular references in merging process (see separated files folder)

## How to test this ticket

- npm run test

## Sample Input and Output (remove if not needed)

Added WSDL test files and tests to verify the process does not throw an infinite loop error

### Checklist:

- [ ] Checked all the config values
- [X] Added relevant unit tests for my code
- [ ] Did not add relevant unit test for my code, by created {{ticket#}} to
      handle in the future

### CODE REVIEW

**Review by VShingala — APPROVED** [review]

File: lib/utils/SchemaBuilderXSD.js [hunk 1]
```diff
@@ -681,7 +682,7 @@ class SchemaBuilderXSD {
       wsdlElement.pattern = element.pattern;
       wsdlElement.enum = element.enum;
 
-      this.getChildren(element, wsdlElement, elements);
+      this.getChildren(getDeepCopyOfObject(element), wsdlElement, elements, targetNamespace);
```
Comment (dhwaneetbhatt) [inline lib/utils/SchemaBuilderXSD.js:685]: @VShingala I had removed this `getDeepCopyOfObject` from here as part of https://github.com/postmanlabs/wsdl-to-postman/pull/196, and it got added back again, I believe, due to resolving conflicts. We should remove this and test it.

**Review by dhwaneetbhatt — CHANGES_REQUESTED** [review]

File: lib/utils/SchemaBuilderXSD.js (hunk 1 above)
Comment (dhwaneetbhatt) [inline lib/utils/SchemaBuilderXSD.js:685]:
```suggestion
      this.getChildren(element, wsdlElement, elements, targetNamespace);
```

Comment (dhwaneetbhatt) [reply lib/utils/SchemaBuilderXSD.js:685]: @VShingala I had removed this getDeepCopyOfObject from here as part of #196, and it got added back again, I believe, due to resolving conflicts. We should remove this and test it.

Comment (VShingala) [reply lib/utils/SchemaBuilderXSD.js:685]: @dhwaneetbhatt Sure, I'll take a look again and see if we are using it elsewhere where not needed.

**Review by dhwaneetbhatt — APPROVED** [review]

**Pull request merged**